python/pyspark/sql/conf.py

```diff
@@ -26,17 +29,17 @@ class RuntimeConfig(object):
     Options set here are automatically propagated to the Hadoop configuration during I/O.
     """
 
-    def __init__(self, jconf):
+    def __init__(self, jconf: JavaObject) -> None:
         """Create a new RuntimeConfig that wraps the underlying JVM object."""
         self._jconf = jconf
 
     @since(2.0)
-    def set(self, key, value):
+    def set(self, key: str, value: str) -> None:
```

Comment:
Seems like this can be more permissive on `value` type (`Union[str, int, bool]`?)